js/store.js

```diff
@@ -1,19 +1,34 @@
 const store = {
+  _websites: null,
+
   init() {
-    this.addListeners();
+    this._read();
+  },
+
+  async _read() {
+    const data = await browser.storage.local.get('websites');
+    this._websites = clone(data.websites);
+
+    return this._websites;
+  },
+
+  async _write(websites) {
+    this._websites = clone(websites);
+
+    return await browser.storage.local.set({ websites });
   },
 
   async getAll() {
-    return await browser.storage.local.get();
+    return await this._read();
   },
 
   async getFirstParty(hostname) {
     if (!hostname) {
       throw new Error('getFirstParty requires a valid hostname argument');
     }
 
-    const storage = await this.getAll();
-    return storage.websites[hostname];
+    const websites = await this.getAll();
+    return websites[hostname];
   },
 
   async getThirdParties(hostname) {
```

Comment:
@princiya, @jonathanKingston I'm okay keeping `hostname`, since it's more general, though does it make sense to have a `getThirdParties` method for any hostname other than `document`?